tofu/geom/_GG02.pyx

```diff
@@ -2620,86 +2620,17 @@ def vignetting(double[:, ::1] ray_orig,
-cdef inline void all_rules(dmode, imode, val_resol, N,
-                           DLs, Lr, los_ind, num_threads=1):
-    # num_los = 1
-    # if dmode=='rel':
-        # N = <int> Cceil(1./val_resol)
-        # if imode=='sum':
-            # coeff_arr = np.empty((N*num_los,), dtype=float)
-            # _st.middle_rule_rel(num_los, N, &DLs[0,0], &DLs[1, 0],
-                                # &dLr[0], &coeff_arr[0], &los_ind[0],
-                                # num_threads=num_threads)
-        # elif imode=='simps':
-            # N = N if N%2==0 else N+1
-            # coeff_arr = np.empty(((N+1)*num_los,), dtype=float)
-            # _st.left_rule_rel(num_los, N,
-                              # &DLs[0,0], &DLs[1, 0], &dLr[0],
-                              # &coeff_arr[0], &los_ind[0],
-                              # num_threads=num_threads)
-        # elif imode=='romb':
-            # N = 2**(int(Cceil(Clog2(N))))
-            # coeff_arr = np.empty(((N+1)*num_los,), dtype=float)
-            # _st.left_rule_rel(num_los, N,
-                              # &DLs[0,0], &DLs[1, 0],
-                              # &dLr[0], &coeff_arr[0], &los_ind[0],
-                              # num_threads=num_threads)
-    # else:
-        # if imode=='sum':
-            # _st.middle_rule_abs_1(num_los, val_resol, &DLs[0,0], &DLs[1, 0],
-                                  # &dLr[0], &los_ind[0],
-                                  # num_threads=num_threads)
-            # ntmp = np.sum(los_ind)
-            # coeff_arr = np.empty((ntmp,), dtype=float)
-            # _st.middle_rule_abs_2(num_los, &DLs[0,0], &los_ind[0],
-                                  # &dLr[0], &coeff_arr[0],
-                                  # num_threads=num_threads)
-            # return coeff_arr, dLr, los_ind[0:num_los-1]
-        # elif imode=='simps':
-            # _st.simps_left_rule_abs(num_los, val_resol,
-                                    # &DLs[0,0], &DLs[1, 0],
-                                    # &dLr[0], &los_coeffs, &los_ind[0],
-                                    # num_threads=num_threads)
-        # else:
-            # _st.romb_left_rule_abs(num_los, val_resol,
-                                   # &DLs[0,0], &DLs[1, 0],
-                                   # &dLr[0], &los_coeffs, &los_ind[0],
-                                   # num_threads=num_threads)
-    pass
-
-
-# TO BE USED IN LOS_calc_signal2 FOR MEMORY MINIMIZATION ?
-cdef inline np.ndarray LOS_get_sample_single(D, u, resol, method, mode):
-    return np.array([])
-    # call _st.all_rules ?
-    # and return k array + scalar effective resolution
-
-
-
-#
-#       TO BE MODIFIED TO RELY ON INLINED single-LOS routine _st.all_rules ?
-#
-def LOS_get_sample(double[:,::1] ray_orig, double[:,::1] us, dL,
-                   double[:,::1] DLs, str dmethod='abs',
+def LOS_get_sample(int num_los, dL, double[:,::1] DLs, str dmethod='abs',
```

Comment:
I realized `ray_orig` and `ray_vdir` (aka `Ds` and `us`) are not used in the function, so took them out...

Reply:
Only `num_los` is actually needed